nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/monitor/SynchronousFileWatcher.java

```diff
@@ -77,21 +77,16 @@ public boolean checkAndReset() throws IOException {
     private boolean checkForUpdate() throws IOException {
         if (resourceLock.tryLock()) {
             try {
-                final StateWrapper wrapper = lastState.get();
+                final Object oldState = lastState.get().getState();
                 final Object newState = monitor.getCurrentState(path);
-                if (newState == null && wrapper.getState() == null) {
+                if (newState == null && oldState == null) {
                     return false;
                 }
-                if (newState == null || wrapper.getState() == null) {
-                    lastState.set(new StateWrapper(newState));
+                lastState.set(new StateWrapper(newState));
```

Comment:
If newState is `null` does this cause issues creating a wrapper around it?

Reply:
I don't think so?  StateWrapper is a simple pojo with getters/setters and no real logic, so I don't think it cares.  This method handles checking for nulls on newState and oldState, so I was happy that NPE wasn't possible.

Reply:
Why the null checks then? Just to be defensive?

Reply:
We don't know what UpdateMonitor implementation is currently in use, and that interface doesn't guarantee that `monitor.getCurrentState(path)` returns NotNull.  So we have to check for nulls before using `newState.equals()` on line 89.